tests/diviner/test_diviner_model_export.py

```diff
@@ -0,0 +1,479 @@
+            key_entries.append(list(value.values())[0])
+        groups.append(tuple(key_entries))
+
+    mlflow.diviner.save_model(diviner_model=grouped_pmdarima, path=model_path)
+    loaded_pyfunc_model = pyfunc.load_pyfunc(model_uri=model_path)
+
+    local_group_pred = grouped_pmdarima.predict_groups(
+        groups=groups,
+        n_periods=10,
+        predict_col="prediction",
+        alpha=0.1,
+        return_conf_int=True,
+        on_error="warn",
+    )
+    pyfunc_conf = pd.DataFrame(
+        {
+            "groups": [groups],
+            "n_periods": 10,
+            "predict_col": "prediction",
+            "alpha": 0.1,
+            "return_conf_int": True,
+            "on_error": "warn",
+        },
+        index=[0],
+    )
+    pyfunc_group_predict = loaded_pyfunc_model.predict(pyfunc_conf)
+
+    pd.testing.assert_frame_equal(local_group_pred, pyfunc_group_predict)
+
+
+def test_diviner_signature_and_examples_saved_correctly(grouped_prophet, diviner_data):
+
+    prediction = grouped_prophet.forecast(horizon=20, frequency="D")
+    signature_ = infer_signature(diviner_data.df, prediction)
+    example_ = diviner_data.df[0:5].copy(deep=False)
+    for signature in (None, signature_):
+        for example in (None, example_):
+            with TempDir() as tmp:
+                path = tmp.path("model")
+                mlflow.diviner.save_model(
+                    grouped_prophet, path=path, signature=signature, input_example=example
+                )
+                mlflow_model = Model.load(path)
+                assert signature == mlflow_model.signature
+                if example is None:
+                    assert mlflow_model.saved_input_example_info is None
+                else:
+                    r_example = _read_example(mlflow_model, path).copy(deep=False)
+                    # NB: datetime values are implicitly cast, so this needs to be reverted.
+                    r_example["ds"] = pd.to_datetime(r_example["ds"], format=DS_FORMAT)
+                    np.testing.assert_array_equal(r_example, example)
+
+
+def test_diviner_load_from_remote_uri_succeeds(grouped_pmdarima, model_path, mock_s3_bucket):
+    mlflow.diviner.save_model(diviner_model=grouped_pmdarima, path=model_path)
+
+    artifact_root = f"s3://{mock_s3_bucket}"
+    artifact_path = "model"
+    artifact_repo = S3ArtifactRepository(artifact_root)
+    artifact_repo.log_artifacts(model_path, artifact_path=artifact_path)
+
+    # NB: cloudpathlib would need to be used here to handle object store uri
+    model_uri = os.path.join(artifact_root, artifact_path)
+    reloaded_model = mlflow.diviner.load_model(model_uri=model_uri)
+
+    pd.testing.assert_frame_equal(grouped_pmdarima.predict(10), reloaded_model.predict(10))
+
+
+def test_diviner_log_model(grouped_prophet, tmpdir):
+
+    old_uri = mlflow.get_tracking_uri()
+    for should_start_run in [False, True]:
+        try:
+            mlflow.set_tracking_uri("test")
```

Comment:
@BenWilson2 Is this the line that creates a test directory?

https://github.com/mlflow/mlflow/pull/5553#discussion_r841800024